File: app/src/main/java/com/bytedance/tiktok/view/SquareWidthImageView.java
package com.bytedance.tiktok.view;

import android.content.Context;
import android.util.AttributeSet;
import androidx.appcompat.widget.AppCompatImageView;

/**
 * create by libo
 * create on 2020/5/21
 * description 高等于宽的imageview
 */
public class SquareWidthImageView extends AppCompatImageView {

    public SquareWidthImageView(Context context) {
        super(context);
    }

    public SquareWidthImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec);  //设置高度始终等于宽度，即为正方形
    }
}

File: app/src/main/java/com/bytedance/tiktok/view/SquareWidthImageView.kt
package com.bytedance.tiktok.view

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * 高等于宽的imageview
 */
class SquareWidthImageView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null
): AppCompatImageView(context, attrs) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec) //设置高度始终等于宽度，即为正方形
    }
}